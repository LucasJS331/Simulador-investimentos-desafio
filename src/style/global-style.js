import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    *{
        box-sizing: border-box;
        padding: 0;
        margin: 0;
        font-family: 'arial';
    }
    html{
        font-size: 62.5%;
        scroll-behavior: smooth;
    }
    body{
        font-size: 1.6rem;
    }
    input {
      display: block;
      outline: none;
      border: none;
      border-bottom: 1px solid black;
      background-color: transparent;
      padding: 0.5rem;
      width: 100%;
    }

`;
