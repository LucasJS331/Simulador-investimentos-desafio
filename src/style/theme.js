export const theme = {
  colors: {
    primary_color: '#E5E8EB',
    secondary_color: '#FF7D48',
    secondary_color_hover: '#E16736',
  },
  media: {
    default: '(max-width: 830px)',
  },
};
