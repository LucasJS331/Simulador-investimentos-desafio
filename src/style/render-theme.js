import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FinantialContext } from '../context/finantial-context';
import { theme } from './theme';

export const renderTheme = (children) => {
  return render(
    <FinantialContext>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FinantialContext>,
  );
};
