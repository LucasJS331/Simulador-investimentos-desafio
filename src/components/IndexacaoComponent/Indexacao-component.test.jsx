import { fireEvent, screen } from '@testing-library/react';
import { Indexacao } from '.';
import { renderTheme } from '../../style/render-theme';

describe('<Indexacao />', () => {
  it('Should render indecacao', () => {
    const { container } = renderTheme(<Indexacao />);
    expect(container).toMatchSnapshot();
  });
  it('Should change option', () => {
    renderTheme(<Indexacao />);
    const fixado_option = screen.getByText(/fixado/i);
    const pre_option = screen.getByText(/pr√©/i);

    fireEvent.click(fixado_option);
    expect(fixado_option).toHaveClass('btn-selected-bg-br10');
    fireEvent.click(pre_option);
    expect(fixado_option).toHaveClass('btn-selected-br10');
    expect(pre_option).toHaveClass('btn-selected-bg');
  });
});
