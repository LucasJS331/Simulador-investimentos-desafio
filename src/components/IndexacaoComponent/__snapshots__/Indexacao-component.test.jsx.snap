// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Indexacao /> Should render indecacao 1`] = `
<div>
  <div
    class="sc-bdvvtL evxNMc"
  >
    <h3
      class="index-title"
    >
      Tipos de Indexação
    </h3>
    <a
      class="btn-selected-bg"
      href="#"
    >
      PRÉ
    </a>
    <a
      class=" btn-selected-br0 "
    >
      PÓS
    </a>
    <a
      class="btn-selected-br10"
    >
      FIXADO
    </a>
  </div>
</div>
`;
